### DISCUSSION TITLE
Can we purge older events to free-up space?

### DISCUSSION THREAD

**Discussion by InfantIvan** [rasmus/EventFlow#944] [Q&A]
We are using eventflow for an integration system. Data retention is just 1 or 2 weeks, but the traffic is high. so we would like to delete the events older than 2 weeks ( those with snapshots). I have 2 questions on that.

1) Can we delete the older events which already have snapshots? eg. Current version of an aggregate is 12 & snapshot version is 10. can we delete the first 5 events ( based on TTL or datestamp)

2) Is there any mechanism to prevent deletion of events without snapshot?

**Comment by rasmus** [comment]
1. You can, but I would recommend that you archive the events in some manner
2. No, there's currently none and its properly hard to do as it depends on the permissions to the underlaying storage
